components/bl-mention-component/src/index.js

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import primereact from './lib/core';
+import { filterSuggestions, getSuggestions, orderFields, stringToList, useTriggers } from './helpers.js';
+import { SuggestionCard } from './suggestion-card.js';
+
+const { Mention } = primereact.mention;
+const { cn } = BackendlessUI.CSSUtils;
+const { __appId: appId, __apiKey: apiKey } = Backendless;
+
+const BLACKLISTED_FIELDS = new Set(['created', '___class', 'ownerId', 'updated', 'objectId']);
+
+export default function MentionComponent({ component, eventHandlers }) {
+  const {
+    trigger, suggestions, field, scrollHeight, autoHighlight, placeholder, delay, autoresize, rows, cols,
+    hideField, classList, style, display,
+  } = component;
+  const { onChange, onFocus, onBlur, onShow, onHide, onSearch } = eventHandlers;
+
+  const [suggestionsMap, setSuggestionsMap] = useState(() => new Map());
+  const [processedSuggestions, setProcessedSuggestions] = useState([]);
+  const triggers = useTriggers(trigger);
+  const [blacklistedFields, setBlacklistedFields] = useState(BLACKLISTED_FIELDS);
+
+  useEffect(() => {
+    suggestions?.forEach(el => setSuggestionsMap(prev => prev.set(el.trigger, el.suggestions)));
+  }, [suggestions]);
+
+  useEffect(() => {
+    if (hideField?.length) {
+      const hidenFields = stringToList(hideField);
+      setBlacklistedFields(new Set([...BLACKLISTED_FIELDS, ...hidenFields]));
+    }
+  }, [hideField]);
+
+  const searchHandler = useCallback(event => {
+    const { trigger: eventTrigger, query } = event;
+
+    const outputList = onSearch({ trigger: eventTrigger, query });
```

Comment:
the `onSearch` event has to return a list of suggestions and inside the event a dev will be able to load it from Backendless or from another storage